engine/step/hook.go

```diff
@@ -0,0 +1,27 @@
+package step
+
+import (
+	"github.com/juju/errors"
+)
+
+// Hook represents the hook data structure
+type Hook struct {
+	Name            string            `json:"name"`
+	Description     string            `json:"description""`
+	LongDescription *string           `json:"long_description,omitempty"`
+	DocLink         *string           `json:"doc_link,omitempty"`
+	Action          Executor          `json:"action"`
+	Results         map[string]string `json:"results"`
```

Comment:
I think we should change this to a map[string]interface{} to match what we have in  `models/task/task.go`